### ISSUE TITLE
Cassette is being created but not persisted

### ISSUE THREAD

**Issue by bencalegari** [vcr/vcr#475]
I have a number of working VCR cassettes but for some reason this one will not save itself once it has been completed. 

When I put a binding in the spec to check, it's telling me it's using a yet to be saved cassette and all appears well.

```
[1] pry(#<RSpec::ExampleGroups::UsersController>)> VCR.current_cassette.name
=> "UsersController/updates a user in infusionsoft on the update action"
[2] pry(#<RSpec::ExampleGroups::UsersController>)> VCR.current_cassette.file
=>".../spec/fixtures/vcr_cassettes/UsersController/updates_a_user_in_infusionsoft_on_the_update_action.yml"
```

When a user is created using factory_girl, an after_create creates that user in Infusionsoft. The test will pass the first time I run it. Every consecutive time it creates more and more actual contacts, causing it to fail. This is what shows up when i ls into the directory where it claims to be creating the cassettes:

```
» ls -la spec/fixtures/vcr_cassettes                                                                       
InfusionsoftContact         User                        user_can_view_personal_data
```

Where did it go? Thanks in advance and please let me know what other information you might need. 

Spec file:

``` ruby
require 'spec_helper'
include InfusionsoftContactFixture

describe UsersController, type: :controller do

  it 'updates a user in infusionsoft on the update action', :vcr do
    user = create(:seller_with_real_infusionsoft_contact)
    sign_in (user)

    infusionsoft_contact_id = user.infusionsoft_contact.id

    put :update, id: user.to_param, user: {id: user.to_param, first_name: 'Updated Name'}

    updated_contact_info = Infusionsoft.contact_find_by_email(user.email, [:FirstName])
    Infusionsoft.contact_delete(infusionsoft_contact_id)

    expect(updated_contact_info).to eql(
      [
        {
          'FirstName' => 'Updated Name'
        }
      ]
    )

    expect(user.reload.first_name).to eql('Updated Name')
  end
end
```

VCR Config:

``` ruby
VCR.configure do |config|
  config.cassette_library_dir = 'spec/fixtures/vcr_cassettes'
  config.configure_rspec_metadata!
  config.default_cassette_options = { :record => :new_episodes }
  config.hook_into :faraday
  config.filter_sensitive_data('<INFUSIONSOFT_API_KEY>') { ENV['INFUSIONSOFT_API_KEY'] }
end
```

**Comment by krainboltgreene** [comment]
Think there's a chance you could make an isolated replication of this problem?

**Comment by bencalegari** [comment]
I'll give it a shot when I get some time!

**Comment by Salet** [comment]
This issue is quite old but I've just ran into a similar behaviour when casette was in use but not persisted. In my case VCR cassettes are inserted with `insert_cassette` method in `before(:each)` block:
http://www.rubydoc.info/github/vcr/vcr/VCR:insert_cassette
And the issue was caused by an application bug which resulted in `eject_cassette` method not being called at the end. 

I know that in the original issue the :vcr tag was used, but I though I will share as it may help someone who rans into this page after Googling, just like I did.
